File: qualify-backend/pkg/analyst.go
package pkg

type Analyst struct {
	User
	Hourly_rate   float64  `json:"hourly_rate"`
	Total_reviews int      `json:"total_reviews"`
	Mean_rating   *float64 `json:"mean_rating"`
}

type AnalystCreateRequest struct {
	Hourly_rate float64 `json:"hourly_rate"`
}

type AnalystProfile struct {
	UserProfile
}

type AnalystSkill struct {
	Analyst_id int `json:"analyst_id,omitempty"`
	Skill_id   int `json:"skill_id"`
}

type AnalystCertification struct {
	Analyst_id       int `json:"analyst_id,omitempty"`
	Certification_id int `json:"certification_id"`
}

type AnalystsResponse struct {
	Analysts []Analyst `json:"analysts"`
	Count    int       `json:"count"`
}

type AnalystResponse struct {
	Analyst Analyst `json:"analyst"`
}

type AnalystProfileResponse struct {
	Analyst_profile AnalystProfile `json:"analyst_profile"`
}

type AnalystSkillResponse struct {
	Analyst_skill AnalystSkill `json:"analyst_skill"`
}

type AnalystSkillsResponse struct {
	Analyst_skills []AnalystSkill `json:"analyst_skills"`
	Count          int            `json:"count"`
}

type AnalystCertificationResponse struct {
	Analyst_certification AnalystCertification `json:"analyst_certification"`
}

type AnalystUpdateRequest struct {
	UserUpdateRequest
	Hourly_rate   *float64 `json:"hourly_rate,omitempty"`
	Total_reviews *int     `json:"total_reviews,omitempty"`
	Mean_rating   *float64 `json:"mean_rating,omitempty"`
}

File: qualify-frontend/types/services/analyst.ts
export interface Analyst {
  id: number;
  name: string;
  email: string;
  phone: string;
  city: string;
  country_code: string;
  country_name: string;
  country_state: string;
  timezone: string;
  hourly_rate: number;
  mean_rating: number;
  total_reviews: number;
  time_created: string;
}

export interface AnalystResponse {
  analyst: Analyst;
}

export interface AnalystsResponse {
  analysts: Analyst[];
  count: number;
}

export interface AnalystUpdateRequest {
  name?: string;
  email?: string;
  phone?: string;
  city?: string;
  country_code?: string;
  country_name?: string;
  country_state?: string;
  timezone?: string;
  hourly_rate?: number;
  mean_rating?: number;
  total_reviews?: number;
}

export interface AnalystProfile {
  user_id?: number;
  picture?: string;
  biography: string;
}

export interface AnalystProfileResponse {
  user_profile: AnalystProfile;
}

export interface ListAnalystsParams {
  name?: string;
  email?: string;
  country?: string;
  country_code?: string;
  country_state?: string;
  city?: string;
  time_zone?: string;
  min_hourly_rate?: number;
  max_hourly_rate?: number;
  min_total_reviews?: number;
  max_total_reviews?: number;
  min_rating?: number;
  max_rating?: number;
  skills?: string;
  sort_by?: string;
  order?: "ASC" | "DESC";
}
